src/core/layout/qgslayoutitemelevationprofile.cpp

```diff
@@ -0,0 +1,421 @@
+/***************************************************************************
+                             qgslayoutitemelevationprofile.cpp
+                             ---------------------------------
+    begin                : January 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayoutitemelevationprofile.h"
+#include "qgslayoutitemregistry.h"
+#include "qgsplot.h"
+#include "qgsmessagelog.h"
+
+///@cond PRIVATE
+class QgsLayoutItemElevationProfilePlot : public Qgs2DPlot
+{
+  public:
+
+    QgsLayoutItemElevationProfilePlot()
+    {
+    }
+
+    void renderContent( QgsRenderContext &rc, const QRectF &plotArea ) override
+    {
+      Q_UNUSED( rc );
+      Q_UNUSED( plotArea )
+    }
+
+  private:
+
+};
+///@endcond PRIVATE
+
+QgsLayoutItemElevationProfile::QgsLayoutItemElevationProfile( QgsLayout *layout )
+  : QgsLayoutItem( layout )
+  , mPlot( std::make_unique< QgsLayoutItemElevationProfilePlot >() )
+{
+
+  //get default layout font from settings
+  const QgsSettings settings;
+  const QString defaultFontString = settings.value( QStringLiteral( "LayoutDesigner/defaultFont" ), QVariant(), QgsSettings::Gui ).toString();
+  if ( !defaultFontString.isEmpty() )
+  {
+//    QFont f = mFormat.font();
```

Comment:
This whole commented out block, should it be taken out?

Reply:
yep, thanks!